### PR TITLE
Add iOS & Android docs for custom GraphQL headers

### PR DESCRIPTION
**Pull request by jamesonwilliams**
Adds documentation to the **Advanced Workflows** page in the GraphQL API documentation for **iOS and Android**.

Explains how to add custom headers to outgoing GraphQL requests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by raphkim — APPROVED** [review]
Android docs look good 🚢 

(iOS doc to be reviewed)

File: docs/lib/graphqlapi/fragments/ios/advanced-workflows/50_interceptor.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+To include custom headers in your outgoing requests, add an `URLRequestInterceptor` to the `AWSAPIPlugin`. Also specify the name of one of the APIs configured in your **amplifyconfiguration.json** file.
+
+<amplify-block-switcher>
+
+<amplify-block name="Listener (iOS 11+)">
```
Comment (drochetti) [inline docs/lib/graphqlapi/fragments/ios/advanced-workflows/50_interceptor.md:5]: maybe we don't need the block switcher in this case? It seems like Listener vs Combine doesn't play a role in this particular API.

**Review by drochetti — APPROVED** [review]
This looks great! Thanks again for putting this together. I left a minor comment, but giving you a _ship it_ regardless. 🚀

File: docs/lib/graphqlapi/fragments/ios/advanced-workflows/50_interceptor.md (hunk 1 above)
Comment (jamesonwilliams) [reply docs/lib/graphqlapi/fragments/ios/advanced-workflows/50_interceptor.md:5]: Fair point! Will simplify to a single snippet.

File: docs/lib/graphqlapi/fragments/ios/advanced-workflows/50_interceptor.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        guard let mutableRequest = nsUrlRequest.mutableCopy() as? NSMutableURLRequest else {
+            throw APIError.unknown("Could not get mutable request", "")
+        }
+        mutableRequest.setValue("headerValue", forHTTPHeaderField: "headerKey")
+        return mutableRequest as URLRequest
+    }
+}
+val apiPlugin = try AWSAPIPlugin()
+try apiPlugin.add(interceptor: CustomInterceptor(), for: "yourApiName")
+try Amplify.addPlugin(apiPlugin)
```
Comment (lawmicha) [inline docs/lib/graphqlapi/fragments/ios/advanced-workflows/50_interceptor.md:16]: I noticed the usage for the api plugin is required to be done after configuring

```suggestion
try Amplify.addPlugin(apiPlugin)
try Amplify.configure()
try apiPlugin.add(interceptor: CustomInterceptor(), for: "yourApiName")
```

Comment (jamesonwilliams) [reply docs/lib/graphqlapi/fragments/ios/advanced-workflows/50_interceptor.md:16]: Thanks @lawmicha!

**Review by lawmicha — APPROVED** [review]
looks good to me after with suggested change

**Pull request merged**